src/wasm/wasm.cpp

```diff
@@ -231,17 +230,24 @@ void If::finalize(Type type_) {
 }
 
 void If::finalize() {
-  type = ifFalse ? Type::getLeastUpperBound(ifTrue->type, ifFalse->type)
-                 : Type::none;
+  if (ifFalse) {
+    // TODO: Calculate a proper LUB.
+    Type::ensureSuperType(std::array<Type, 2>{{ifTrue->type, ifFalse->type}},
```

Comment:
Is the `std::array` necessary? A shame...

Reply:
Yes, unfortunately we need to assign a type to the braced initializer list so `ensureSuperType` knows what kind of iterators it is dealing with.